pupil_src/shared_modules/calibration_routines/gaze_mappers.py

```diff
@@ -76,68 +76,128 @@ class Binocular_Gaze_Mapper_Base(Gaze_Mapping_Plugin):
 
     def __init__(self, g_pool):
         super().__init__(g_pool)
+        self._cfg = {'min_pupil_confidence': 0.6,
+                     'temporal_cutoff': 0.03,
+                     'timestamp_reset_threshold': 5.0,
+                     'process_low_confidence': True,
+                     'sample_cutoff': 10}
+        self._state = {'cache': deque(),
+                       'cache_ts': deque(),
+                       'cache_id': deque(),
+                       'cache_conf': deque()}
+
+    def _reset(self):
+        self._state['cache'].clear()
+        self._state['cache_ts'].clear()
+        self._state['cache_id'].clear()
+        self._state['cache_conf'].clear()
 
-        self.min_pupil_confidence = 0.6
-        self._caches = (deque(), deque())
-        self.temportal_cutoff = 0.3
-        self.sample_cutoff = 10
+    def add_menu(self):
+        super().add_menu()
+        self.menu.append(ui.Slider("min_pupil_confidence", self._cfg, step=0.01, min=0.0, max=1.0,
+                                   label="Min conf. for binocular",))
+        self.menu.append(ui.Slider("temporal_cutoff", self._cfg, step=0.002, min=0.0, max=0.2,
+                                   label="Max binocular time diff. (s)",))
+        self.menu.append(ui.Slider("sample_cutoff", self._cfg, step=1, min=2, max=20,
+                                   label="Max queue length", ))
 
     def map_batch(self, pupil_list):
-        current_caches = self._caches
-        self._caches = (deque(), deque())
+        backup_state = self._state
+        self._reset()
```

Comment:
Unfortunately, this also resets `backup_state`  since it's `deque`s are only referenced, not deep-copied.

Reply:
Thanks. I don't use `deque` very much. It's fixed in the new commit.